### PR TITLE
Switch to Eclipse Temurin as base image

### LINKED ISSUE #194: AdoptOpenJDK images deprecated
adoptopenjdk images have been deprecated in favor of eclipse-temurin images. We could switch to that image, but for legal reasons, they won't provide OpenJ9 images, these must be provided by IBM directly. Currently, ibmjava images are not deprecated and there is no plan to do so, but these also don't offer OpenJ9 images. I don't currently have a solution for where to get OpenJ9 images in the future.

As for the non-OpenJ9 images, we can either go back to the openjdk images (which will no longer force usage of Oracle Linux I believe) or switch to the eclipse-temurin images.  I'm currently more inclined to choose Termurin as it's more likely to expand it's offering and I think represents a bit more of a growing industry standard.

Thoughts on this are welcome.

### PR DESCRIPTION
**Pull request by keeganwitt**
Closes #194

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -11,27 +11,24 @@ jobs:
...
         env:
-          jvm: ${{ matrix.jvm }}
           jvmType: ${{ matrix.jvmType }}
           version: ${{ matrix.version }}
         run: |
-          cd "${jvm}/${jvmType}${version}"
-          docker pull "adoptopenjdk:${version}-${jvmType}-${jvm}"
-          docker build --tag "gradle:${jvmType}${version}-${jvm}" .
+          cd "${jvmType}${version}"
+          docker pull "adoptopenjdk:${version}-${jvmType}"
```
Comment (mleegwt) [inline .github/workflows/ci.yaml:25]: Should this be temurin also?

Comment (keeganwitt) [reply .github/workflows/ci.yaml:25]: Yea. The jdk vs jre bit isn't right yet either.

I'm not committed to making this move yet. It seems like probably the best path forward, but there are some complications (see linked issue).

**Comment by keeganwitt** [comment]
Temurin doesn't (and doesn't plan to) offer JRE image except Java 8.  Originally I had included these thinking they might be useful for those using Gradle for native compilation.  But now it seems like it's more confusion and not clearly a major win in disk size.  If I recall correctly, there's not really a specification for what would be included in a JRE.  The only JRE I've seen shipped is Java 8.

So I've also dropped those.

**Comment by keeganwitt** [comment]
Actually, I misread the conclusion of that GH issue -- They do plan on making JRE images. 8 and 11 are out now, and presumably they are working on 17. Still, I'm thinking they may not make sense for this image. Please speak up if you think differently.

**Pull request merged**